### PR TITLE
[docs] facedetector and barcodeScanner deprecations

### PR DESCRIPTION
**Pull request by alanjhughes**
# Why
Closes ENG-11241
`expo-face-detector` and `expo-barcode-scanner` are now deprecated. 

# How
Add deprecation notices to these packages

# Test Plan
Screenshot 2024-01-27 at 11 21 39
Screenshot 2024-01-27 at 10 46 34

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: docs/pages/versions/unversioned/sdk/bar-code-scanner.mdx [hunk 1]
```diff
@@ -19,6 +19,8 @@ import {
 import { AndroidPermissions, IOSPermissions } from '~/components/plugins/permissions';
 import { PlatformTags } from '~/ui/components/Tag';
 
+> **warning** **Deprecated:** This module will no longer be available in SDK 51. Use [expo-camera](camera-next.mdx) instead.
```
Comment (Simek) [inline docs/pages/versions/unversioned/sdk/bar-code-scanner.mdx:22]:
```suggestion
> **warning** **Deprecated:** This module will no longer be available in SDK 51. Use [expo-camera/next](camera-next) instead.
```

Linking nit.

Also, wdyt about stating explicitly that we recommend "Next" version of the package (at least until the next SDK release)?

Comment (alanjhughes) [reply docs/pages/versions/unversioned/sdk/bar-code-scanner.mdx:22]: Yeah, that makes sense. Is the link enough, or do we need something like
```markdown
We recommend using expo-camera/next which has barcode scanning built in instead.
```

Comment (Simek) [reply docs/pages/versions/unversioned/sdk/bar-code-scanner.mdx:22]: Great idea. Adding a context or broader information how/why is always appreciated.

Comment (alanjhughes) [reply docs/pages/versions/unversioned/sdk/bar-code-scanner.mdx]: Updated. Thanks for reviewing @Simek 👍

**Review by Simek — APPROVED** [review]
Thanks for adding those notes! 👍

**Pull request merged**